demo_nodes_cpp/src/services/add_two_ints_client_async.cpp

```diff
@@ -27,51 +28,85 @@ using namespace std::chrono_literals;
 void print_usage()
 {
   printf("Usage for add_two_ints_client app:\n");
-  printf("add_two_ints_client [-t topic_name] [-h]\n");
+  printf("add_two_ints_client [-s service_name] [-h]\n");
   printf("options:\n");
   printf("-h : Print this help function.\n");
   printf("-s service_name : Specify the service name for this client. Defaults to add_two_ints.\n");
 }
 
-int main(int argc, char ** argv)
+class ClientNode : public rclcpp::Node
 {
-  rclcpp::init(argc, argv);
+public:
+  explicit ClientNode(const std::string & service_name)
+  : Node("add_two_ints_client")
+  {
+    client_ = create_client<example_interfaces::srv::AddTwoInts>(service_name);
+    auto on_timer =
+      [this]() -> void {
+        timer_->cancel();  // We only want to make a single request.
+        this->make_request();
+      };
+    timer_ = create_wall_timer(0s, on_timer);  // This will trigger once spin is called on the node.
```

Comment:
@Karsten1987 and I spoke about this and it should be fine to just call `make_request` straight away here (should rename the method to `start_async_request` or something) since both approaches just queue something for once the node gets spun on.. I'll give it a shot

Reply:
simplified in a010a4dc9021fc658acacbde10632d2840ec3629 @mikaelarguedas fyi (you also mentioned this was unnecessarily complicated FWIU)